### PR TITLE
[Sparse] neg and inv

### PR DESCRIPTION
**Pull request by mufeili**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/mock_sparse2/unary_op_diag.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    DiagMatrix(val=tensor([-0., -1., -2.]),
+    shape=(3, 3))
+    """
+    return diag(-D.val, D.shape)
+
+
+def inv(D: DiagMatrix) -> DiagMatrix:
+    """Return the inverse of the diagonal matrix.
+
+    This function only supports square matrices with nonzero values of shape
```
Comment (jermainewang) [inline python/dgl/mock_sparse2/unary_op_diag.py:30]: ... with scalar nonzero values.

**Review by jermainewang — APPROVED** [review]

File: tests/pytorch/mock_sparse2/test_unary_op_sp.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import pytest
+import torch
+import sys
+
+import backend as F
+
+from dgl.mock_sparse2 import create_from_coo
+
+# FIXME(issue #4818): Skipping tests on win.
```
Comment (frozenbugs) [inline tests/pytorch/mock_sparse2/test_unary_op_sp.py:9]: `TODO(#4818):`

Comment (mufeili) [reply tests/pytorch/mock_sparse2/test_unary_op_sp.py:9]: updated in #4997
consider documenting it somewhere

Comment (frozenbugs) [reply tests/pytorch/mock_sparse2/test_unary_op_sp.py:9]: https://docs.google.com/document/d/1NGXPg4U67ZpMDVxxkb-SprEx11d8zqNB_0cLweTL90Y/edit#bookmark=id.rhioqnfrsekp

Comment (frozenbugs) [reply tests/pytorch/mock_sparse2/test_unary_op_sp.py:9]: Yes, I linked it to https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit#heading=h.4p68oec5ndrr, which is linked to PR template. Most important is we all be aware of this, and point it out in PR review. I don't expect every PR author read all the details when they code.

**Pull request merged**